e2e_playwright/st_fragments_nested_test.py

```diff
@@ -39,29 +95,96 @@ def test_full_app_rerun(app: Page):
     expect(app.get_by_test_id("stMarkdown").nth(1)).not_to_have_text(
         outer_fragment_text
     )
-    expect(app.get_by_test_id("stMarkdown").last).not_to_have_text(inner_fragment_text)
+    expect(app.get_by_test_id("stMarkdown").nth(2)).not_to_have_text(
+        inner_fragment_text
+    )
+    expect(app.get_by_test_id("stMarkdown").last).not_to_have_text(inner_fragment2_text)
 
 
 def test_outer_fragment_rerun(app: Page):
-    outside_fragment_text, outer_fragment_text, inner_fragment_text = get_uuids(app)
+    (
+        outside_fragment_text,
+        outer_fragment_text,
+        inner_fragment_text,
+        inner_fragment2_text,
+    ) = _get_uuids(app)
 
-    click_button(app, "rerun outer fragment")
+    _rerun_outer_fragment(app)
 
     # We reran the outer fragment, so the UUID outside of the fragments should stay
     # constant, but the other two should have changed.
     expect(app.get_by_test_id("stMarkdown").first).to_have_text(outside_fragment_text)
     expect(app.get_by_test_id("stMarkdown").nth(1)).not_to_have_text(
         outer_fragment_text
     )
-    expect(app.get_by_test_id("stMarkdown").last).not_to_have_text(inner_fragment_text)
+    expect(app.get_by_test_id("stMarkdown").nth(2)).not_to_have_text(
+        inner_fragment_text
+    )
+    expect(app.get_by_test_id("stMarkdown").last).not_to_have_text(inner_fragment2_text)
 
 
 def test_inner_fragment_rerun(app: Page):
-    outside_fragment_text, outer_fragment_text, inner_fragment_text = get_uuids(app)
+    (
+        outside_fragment_text,
+        outer_fragment_text,
+        inner_fragment_text,
+        inner_fragment2_text,
+    ) = _get_uuids(app)
+
+    _rerun_inner_fragment(app)
+
+    # We reran the inner fragment. Only that corresponding UUID should have changed.
```

Comment:
nit: may want to reword this comment slightly (and the similar one below) to differentiate between the two inner fragments

Reply:
ah great point, I have updated all the comments in the test file